---
title: 'Mutations in SARS-CoV-2 Spike Protein and its Role in Transmissibility'
author: "Lewis Hendrianto"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: pdf_document
csl: bioinformatics.csl
bibliography: references.bib
params:
  gff_file_path: "data/metadata/sars_refgenome_annotation.gff"
  vcf_dir_path: "data/11_vcf_output_for_R"
  sra_runtable_path: "data/00_sra_runtable/filtered_sra_table.txt"
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      error = FALSE)
```

# Background and Overview

This report examined how single nucleotide polymorphisms (SNPs) in different strains of SARS-CoV-2 affects their permeability and transmissibility. It has been established through the taken BioProject that the substitution S:655Y enhances replication of the disease, transmitting more efficiently than variants. The primary goal of this analysis were to identify the SNPs present in the variants of the disease and analyze how might the SNPs enhance the transmissibility of the variants. The approach was conducted through a series of bash scripts and Rstudio code. The bash scripts ultimately converted the metadata for the SRA run table to vcf files, which were made accessible to interpret. The analysis was possible through the data contained in the vcf file.

(INSERT FINDINGS) Mock Findings: state what the ref nuc and alt nuc is. State what chrom pos and what the snp is. state the quality of score of each chrom pos.

maybe add as of march 22 gamma variant died (maybe got out competed maybe add how gamma was prevalent in brazil and associated with increased death risk and severity in younger ages. flow - snp lead to increased permeability to increased death

# Methods

See the set of tutorials on the [vcfR package website](https://knausb.github.io/vcfR_documentation/index.html).

You may also want to use any of a range of different COVID data packages and data sources:

-   <https://kjhealy.github.io/covdata/>
-   <https://github.com/como-ph/oxcovid19>
-   <https://ropensci.org/blog/2020/10/20/searching-medrxivr-and-biorxiv-preprint-data/>
-   <https://covidtracking.com/data/api>
    -   `readr::read_csv("https://api.covidtracking.com/v1/states/daily.csv")`

## Subsections are ok too

# Results and Discussion

```{r load-packages-and-functions}
library("vcfR")
library("ggplot2")
library("dplyr")
library("ggthemes")
library("tidyr")

# this loads all function files in the code/functions directory
function_paths <- list.files("code/functions", full.names = TRUE)
purrr::walk(function_paths, source)
```

```{r load-vcf-data}
# load in, tidy, and stack all of the VCF files in the target directory
# these are the output of the bash pipeline
stacked_vcfs <- parse_tidy_and_stack_vcfs(
  vcf_dir_path = params$vcf_dir_path)

# load in the gff file with genome annotations for SARS-CoV-2
gff <- read_gff(gff_file_path = params$gff_file_path)

# pull out just the named genes from the annotation file
gene_table <- extract_genes_from_gff(annotation_object = gff)

# combine the stacked vcf table with the gene names and the metadata from
# the sra runtable file
vcf_with_metadata <- add_genes_metadata_to_vcfstack(
  sra_runtable_path = params$sra_runtable_path,
  stacked_vcf = stacked_vcfs,
  cleaned_genes_table = gene_table)
vcf_with_metadata$qual <- as.integer(vcf_with_metadata$qual)
colnames(vcf_with_metadata) <- gsub(" ", "_", colnames(vcf_with_metadata))

```

# Figures

```{r dot-plot}
# create a dot plot mapping out all of the samples collection date and
# their quality score
dot_plot <- ggplot(vcf_with_metadata[which(vcf_with_metadata$qual>150),],
                       aes(x = Collection_Date,
                           y = qual)) +
  geom_point(col = "black", size = 1) +
  scale_x_date(date_labels = "%Y-%b-%d",date_breaks  = "1 month") +
  labs(x = "Date of sample collection",
       y = "Quality score of sample",
       title = "The collection date and quality score of each sample") +
    theme(axis.text.x = element_text(angle = 0, size = 7)) +
  theme(axis.text.y = element_text(angle = 0, size = 7)) +
  theme(plot.title = element_text(hjust = 0.5))

dot_plot
```

**Figure 1** The collection date of the samples and their quality scores

```{r SNP-quality-plot}
# create a flipped bar plot showing which SNPs are quality SNPs and which
# should be excluded
bar_plot <- ggplot(data = vcf_with_metadata,
                    aes(x = alt,
                        y = qual)) +
  geom_col(position = "dodge") +
  labs(x = "DNA base",
       y = "Quality score",
       title = "Quality of SNP") +
    theme(axis.text.x = element_text(angle = 0, size = 7)) +
  theme(axis.text.y = element_text(angle = 0, size = 7)) +
  theme(plot.title = element_text(hjust = 0.5)) +
  scale_x_discrete(guide = guide_axis(n.dodge = 1.75)) +
  coord_flip()

bar_plot
```

**Figure 2** The quality and confidence of the SNPs in the BioProject

# Tables

```{r sample-bases-table}
# create a table showing the samples and what the isolation source is, and how
# many bases they have

vcf_with_metadata %>%
  select(sample, Isolation_source, Bases) %>% unique() %>%
  knitr::kable(col.names = c("Sample",
                             "Isolation source",
                             "Number of Bases"), row.names = FALSE,
               align = "lll")

```

**Table 1** The test samples and where they originated from

```{r snp-table}
# create a table mapping out where the SNP occurs

stacked_vcfs %>%
  select(pos, ref, alt) %>% unique() %>%
  knitr::kable(col.names = c("Chromosome position",
                             "Ref. Nucleotide",
                             "SNP"),
               row.names = FALSE, align = "lcc")

```

**Table 2**: Where each SNP occurs

```{r date-table}
# create a table showing when the samples were collected and released

vcf_with_metadata %>%
  arrange(vcf_with_metadata$Collection_Date) %>%
  select(Collection_Date, ReleaseDate) %>% unique() %>%
  knitr::kable(col.names = c("Collection Date",
                             "Release Date"),
               row.names = FALSE, align = "ll")

```

**Table 3**: The samples collection date and release date

# Sources Cited
