---
title: 'Mutations in SARS-CoV-2 and its Effect in Viral Replication and Entry'
author: "Lewis Hendrianto"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: pdf_document
csl: bioinformatics.csl
bibliography: references.bib
params:
  gff_file_path: "data/metadata/sars_refgenome_annotation.gff"
  vcf_dir_path: "data/11_vcf_output_for_R"
  sra_runtable_path: "data/00_sra_runtable/filtered_sra_table.txt"
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      error = FALSE)
```

# Background and Overview

  The World Health Organization (WHO) declared coronavirus 19 (COVID-19), a pandemic on March 11, 2020. (https://www.yalemedicine.org/news/covid-timeline) Severe acute respiratory syndrome coronavirus 2, also known as SARS-CoV-2, is a virus from the coronavirus family. SARS-CoV-2 causes this disease. COVID-19 is a respiratory disease primarily transmitted through aerosols. Preventative and therapeutic approaches are in development to combat this virus, and an mRNA vaccine has become widely available for the public. (https://www.frontiersin.org/articles/10.3389/fcimb.2020.587269/full)
  Before understanding the function of the virus, it is essential to understand the structure of a virus. In general, each virus contains a viral genome that resides in a shell called the capsid. Additionally, the coronavirus contains a lipid membrane envelope that surrounds the capsid. Spike glycoproteins surround the lipid envelope, functioning as a gateway to enter the target host. The spike proteins bind to a specific receptor to the host, for SARS-CoV-2, its target is the Angiotensin-converting enzyme 2 (ACE2). Upon binding, the virus will permeate the host cell’s membrane, inserting the viral genome. The viral genome gets replicated and expressed, producing viral proteins infect other cells ultimately causing COVID-19. (https://www.khanacademy.org/science/biology/biology-of-viruses/virus-biology/a/intro-to-viruses) Moreover, the spike proteins could undergo mutations, affecting is permeability and transmissibility towards the target host. Consequently, the SARS-CoV-2 has faced several mutations, acquiring SNPs in multiple positions. With the data collected through BioProject PRJNA793894, it has been identified that the S:665Y substitution causes these effects. This report will take the data and examine how the SNPs affected the viral entry and replication of the newly obtained variants. On November 8, 2021 the first confirmed sample of the Omicron variant is discovered in South Africa. In less than a month, the variant of concern (VOC) made its way across the world to the United States on December 1st, 2021. Rapidly, the Omicron variant became one of the most dominant strains in the world because of the SNP in the spike protein.(https://health.ucdavis.edu/coronavirus/covid-19-information/omicron-variant) As of November 2022, Omicron is responsible for over 97% of COVID-19 transmissions according to the Centers for Disease Control and Prevention. As a result, this VOC takes in account for the majority of COVID-19 cases and deaths since its emergence. (https://covid.cdc.gov/covid-data-tracker/#variant-proportions)
  The primary goal of this analysis were to identify the position of each SNP and interpret how they enhanced the transmissibility of this VOC compared to the original variant. Although the data collected from the biosample was from June 2021 through November 2021, the report discovered how the SNP has impacted the United States in recent times. The approach of the report was accomplished through a combination of bash scripts, R code, and Perl code. The provided code took the downloaded BioProject’s SRA RunTable metadata and executed several ramifications. The 89 Illumina samples in the RunTable was deconstructed, trimmed, and polished to produce VCF files that exposed the SNPs, quality scores, and other interpretable data that allowed a close analysis. Finally, a Makefile compiled the analysis and produced the report as a PDF.

# Methods

  The BioProject was selected and extracted through the National Center for Biotechnology Information (NCBI) website with parameters to isolate COVID-19 data. For an appropriate analysis, the chosen dataset was preferred to have various metadata and SRA experiments. Additionally, the SRA experiments had to be Illumina data for analysis approach to work. Upon choosing an adequate BioProject to work with, the accession number was put through the NCBI’s SRA Run Selector. The metadata of the SRA RunTable was downloaded to begin the analysis. The analysis approach used a series of bash scripts that dependently functions with the preceding scripts. For this particular dataset, there were Oxford Nanopore data, and a script was created to remove the Nanopore data. In total, the fifteen scripts processed the RunTable for a complete analysis. The scripts were compiled in a Makefile, accessible through the terminal. Upon the input `make`, the parsing begins. Also, an R Markdown file was needed to produce the final report with functional figures/tables and text. 
  The first script in the pipeline set up the anticipated directories was ran. The script produced thirteen total directories, stored in two places. A `/data/` folder was made along with a `data/` folder. Eleven of the directories were stored in the first folder, and the rest in the latter. After the directory set up, the series of scripts first used a fasterq-dump. The fasterq-dump from the SRA tools software will download the fastq files for each SRA run ID from the RunTable file, parsing it. The script also removed the reverse reads and stored it in a directory for raw fastq files. A second script downloaded the reference genome for the fasta file with the `curl` R code and stored it in its own directory. Following, a separate script will download the annotation gff, using `curl` and extracting the genome annotation with `gunzip`, storing it in its directory. Taking the fastq files, a script will run `fastqc` on the files. The processed fastq files are then trimmed with a script that used `TrimmomaticSE`, throwing out the bad sequences. Aside from the trimmed fastq files, the next script will take the reference genome and index the file for Burrows-Wheeler Aligner (BWA). A script that runs BWA took the given files and aligns the each sample’s reads as inputs to the reference genome, using `bwa mem`, storing the output as a .sam file. The .sam files are converted to .bam files with the next script, using the `samtools view` command. The .bam files are sorted with another script that used the `samtools sort -o` command, storing it in a new directory. The sorted bamfiles undergone a flagstat script. The script ran `flagstats` on the sorted files with the `samtools flagstat` command, extracting information from the sorted files, storing the stats in as a text file in a directory. The next script in the pipeline called the variants in the sorted bam files, and calculated the read coverage of the chromosomal positions in the entire genome which used the `bcftools mpileup` command. The script ultimately extracted information on the read coverage for each base, storing it in the made directory. Subsequently, another script took the bcf files, and exposed the SNPs for each file, using the `bcftools call –ploidy 1` command, saving it as a .vcf file. The next script used a perl script to filter out the short variants and SNPs on the .vcf file. The script used `vcfutils.pl varFilter` to execute the function. These series of scripts completed the whole analysis of the single RunTable, designating each script to save their files in a separate directory. The last remaining bash script drove the rendering of an Rmarkdown file. It required four arguments to run: the Rmd file, path to the gff annotation, path to the directory of processed vcf files, and the SRA RunTable with metadata.
  With a simple command of `make` in the terminal, the Makefile was set into motion, executing all of the scripts. In the end, a pdf document was produced displaying the report of the SARS-CoV-2 analysis.

# Results

The BioProject collected 89 different samples, from four different isolates at the Icahn School of Medicine at Mount Sinai, located in New York, United States. The isolates originated from a nasal wash or nasal turbinate from *Mesocricetus auratus* and viral supernantants with pneumocyte infected cells and without (Table 1). The samples were collected in the months of June 2021 and November 2021. Although a number of the collection dates were before the identification of Omicron, the same SNPs were present in the collection dates during Omicron’s emergence. Shortly after, the BioProject was released on January 3rd, 2021, past Omicron’s first invasion into the United States. Regarding the spike protein gene *S*, the SNP from nucleotide C to T remained conserved in both month’s of the collection date, occurring at the same chromosomal position 23525. Correspondingly, the other SNPs present and the SNP for the *ORF8* gene also remained conserved (Table 2). Parsing through the SRA runtable with the bash scripts, it was determined that the vast majority of SNPs in the sample were reliable and certain. The vast majority of the samples possessed SNPs that were confirmed. The samples almost consistently had a score greater than 220, with the some scoring in the range of 190 to 220 (Fig. 1, Fig. 2). For the latter, they were ruled out of the analysis due to the poor quality scores of 134 or lower. The poor SNPs mainly occurred at position 23525, with one changing to base pair A at 18883, and a change to C at 29737 (Table 3). It has been established that there are four certain SNPs that have occurred in SARS-CoV-2. The most common SNP changes the nucleotide C to T, and the other changing from T to C.However, only one SNP affects the spike protein’s gene, at 23525. As for the other affected gene, the 28144 SNP of base pair C is shown to alter the ORF8 gene. SNPs at position 8782 and 18060 contains the same SNP as the spike protein SNP, but it is said to not affect any particular gene (Table 4). It has been identified that the SNPs are found in the Omicron variant. As a result, the deaths starting from December 1st, 2021, the day Omicron was identified in the united states, were recorded until November 23, 2022. In the time span, there has been 292, 973 confirmed deaths in the United States due to COVID-19 (Fig. 3). Although the deaths do not explicitly identify the variant to be the cause of death, we should consider that over 97% confirmed cases of COVID-19 are from Omicron since its emergence.

# Discussion

  With the SNP data collected by the Icahn School of Medicine, there is a couple of theories that revolve around the Omicron variant and it’s spike protein. Following the timeline of Omicron, the variant was first identified in South Africa on November 8, 2021. The first detected of Omicron in the United States was December 1st, 2021. (https://en.wikipedia.org/wiki/Timeline_of_the_SARS-CoV-2_Omicron_variant) However, the first recordings of the SNPs were on June 8, 2021, and ending on November 11, 2021. There is a speculation that the Omicron variant has been around since the beginning of June due to the exact SNPs being present in both collection dates. Nonetheless, some SNPs may have been missed in the BioProject that differentiated the integrity of the SNPs, but it is worth noting that the SNPs remained conserved on both collection periods. Moreover, there were four quality SNPs that were identified in the analysis, but only one of the SNPs directly affected the gene responsible for the spike protein, *S*. The other affected gene was *ORF8*. The *ORF8* gene encodes for it’s protein, ORF8. According to a published research paper, SARs-CoV-2 uses the protein to change MHC-I expression, affecting the cell’s immune response. (https://doi.org/10.1073/pnas.2024202118) The MHC-I protein, or major histocompatibility protein 1, is responsible for aiding the immune system in recognizing foreign substances. Specifically, the protein binds to the pathogens and presents them to T cells, as a result, the immune system eliminates the pathogen. (https://www.ncbi.nlm.nih.gov/books/NBK27156/) Therefore, the presumption that mutations in *ORF8* gene disrupts the protein’s opsonization ability, allowing viral replication uncontested. Prior to replication, the virus must find its target. The virus accomplishes this through its spike protein. For SARS-CoV-2, the spike protein binds to the human Angiotensin-Converting Enzyme 2 (ACE2) receptor. The formation of the S protein-ACE2 complex allows the virus to gain entry to the host cell and integrate its viral genome. It has been established that SARS-CoV-2 has a strong binding affinity to the receptor, even stronger than the original SARS-CoV. (https://www.frontiersin.org/articles/10.3389/fmolb.2020.591873/full)
  Conceivably, the SNP in the S protein increases the binding affinity for the ACE2 receptor. In addition, the two SNPs that do not reside within the genes, linked SNPs (or indicative SNPs) may play a role in the enhancement. According to the University of Utah, the linked SNPs do not affect a protein’s function, but they correspond to a certain disease or drug response. The linked SNPs are said to be hallmarks, or genetic markers of their disease. (https://learn.genetics.utah.edu/content/precision/snips#:~:text=SNPs%20are%20divided%20into%20two,for%20getting%20a%20certain%20disease.) Thus, the causative SNPs, the polymorphisms that affect the genes, created a SARS-CoV-2 variant that can permeate and replicate more effectively than others. As a direct consequence, the Omicron dominates in infection rate and death rate. https://www.pbs.org/newshour/health/omicron-has-caused-higher-increase-in-u-s-daily-death-count-than-delta-variant

```{r load-packages-and-functions}
library("vcfR")
library("ggplot2")
library("dplyr")
library("ggthemes")
library("tidyr")
library("readr")

# this loads all function files in the code/functions directory
function_paths <- list.files("code/functions", full.names = TRUE)
purrr::walk(function_paths, source)
```

```{r load-vcf-data}
# load in, tidy, and stack all of the VCF files in the target directory
# these are the output of the bash pipeline
stacked_vcfs <- parse_tidy_and_stack_vcfs(
  vcf_dir_path = params$vcf_dir_path)

# load in the gff file with genome annotations for SARS-CoV-2
gff <- read_gff(gff_file_path = params$gff_file_path)

# pull out just the named genes from the annotation file
gene_table <- extract_genes_from_gff(annotation_object = gff)

# combine the stacked vcf table with the gene names and the metadata from
# the sra runtable file
vcf_with_metadata <- add_genes_metadata_to_vcfstack(
  sra_runtable_path = params$sra_runtable_path,
  stacked_vcf = stacked_vcfs,
  cleaned_genes_table = gene_table)
vcf_with_metadata$qual <- as.integer(vcf_with_metadata$qual)
colnames(vcf_with_metadata) <- gsub(" ", "_", colnames(vcf_with_metadata))
covid_tabular_data <- read_csv(file = "data/raw_data/covid_tabular_data.csv")

```

# Figures

```{r dot-plot}
# create a dot plot mapping out all of the samples collection date and
# their quality score
dot_plot <- ggplot(vcf_with_metadata[which(vcf_with_metadata$qual > 150), ],
                       aes(x = Collection_Date,
                           y = qual)) +
  geom_point(col = "black", size = 1) +
  scale_x_date(date_labels = "%Y-%b-%d", date_breaks  = "1 month") +
  labs(x = "Date of sample collection",
       y = "Quality score of sample",
       title = "The collection date and quality score of each sample") +
    theme(axis.text.x = element_text(angle = 0, size = 7)) +
  theme(axis.text.y = element_text(angle = 0, size = 7)) +
  theme(plot.title = element_text(hjust = 0.5))

dot_plot
```

**Figure 1**: The collection date of the samples and their quality scores

```{r bar-plot}
# create a flipped bar plot showing SNPs quality score
bar_plot <- ggplot(data = vcf_with_metadata,
                    aes(x = alt,
                        y = qual)) +
  geom_col(position = "dodge") +
  labs(x = "DNA base",
       y = "Quality score",
       title = "Quality of SNP") +
    theme(axis.text.x = element_text(angle = 0, size = 7)) +
  theme(axis.text.y = element_text(angle = 0, size = 7)) +
  theme(plot.title = element_text(hjust = 0.5)) +
  scale_x_discrete(guide = guide_axis(n.dodge = 1.75)) +
  coord_flip()

bar_plot
```

**Figure 2**: The SNPs quality score

```{r time-line-plot}
# create a time-line plot that shows the total deaths from
# December 2021 until November 2022
time_plot <- ggplot(data = covid_tabular_data,
                    aes(x = Date)) +
  geom_line(aes(y = Total_Deaths)) +
  geom_point(aes(y = Total_Deaths)) +
  labs(x = "Date",
       y = "Number of Deaths") +
  scale_x_date(date_breaks = "1 month") +
  theme(axis.text.x = element_text(angle = 60, size = 5, hjust = 1)) +
  scale_y_continuous(labels = scales::comma)

time_plot
```

**Figure 3**: The cumulative deaths since the emergence of Omicron in the United States

# Tables

```{r sample-source-location-table}
# create a table showing the isolation source of the sample, and where the
# sample was collected
vcf_with_metadata %>%
  select(sample, Isolation_source, geo_loc_name) %>%
  unique() %>%
  knitr::kable(col.names = c("Sample",
                             "Isolation source",
                             "Location"),
                              row.names = FALSE,
                              align = "lll")

```

**Table 1**: The Isolation source of the samples and collection location

```{r date-table}
# create a table showing when the collection and release date of the sample
# and the position of the SNPs and affected gene
vcf_with_metadata %>%
  arrange(vcf_with_metadata$Collection_Date) %>%
  select(Collection_Date, ReleaseDate, alt, gene, pos) %>%
  unique() %>%
  knitr::kable(col.names = c("Collection Date",
                             "Release Date",
                             "SNP",
                             "Gene",
                             "Position"),
                              row.names = FALSE,
                              align = "ll")

```

**Table 2**: The conserved SNPs and their two collection time frames

```{r bad-SNP-table}
# create a table that shows the SNPs with bad quality scores to rule out of
# the analysis
vcf_with_metadata[which(vcf_with_metadata$qual < 150), ] %>%
  arrange((vcf_with_metadata[which(vcf_with_metadata$qual < 150), ])$pos) %>%
  select(pos, alt, qual) %>%
  unique() %>%
  knitr::kable(col.names = c("Chromosome position",
                             "SNP",
                             "Quality score"),
                              row.names = FALSE,
                              align = "lcc")

```

**Table 3**: The SNPs that are excluded in the analysis

```{r snp-table}
# create a table mapping out where the SNP occurs and the affected gene
vcf_with_metadata %>%
  select(pos, ref, alt, gene) %>%
  unique() %>%
  knitr::kable(col.names = c("Chromosome position",
                             "Ref. Nucleotide",
                             "SNP",
                             "Gene affected"),
                              row.names = FALSE,
                              align = "lccc",
               na.rm = TRUE)
```

**Table 4**: The SNPs and their reference nucleotide, and the affected gene

# Sources Cited
