---
title: 'Mutations in SARS-CoV-2 Spike Protein and its Role in Transmissibility'
author: "Lewis Hendrianto"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: pdf_document
csl: bioinformatics.csl
bibliography: references.bib
params:
  gff_file_path: "data/metadata/sars_refgenome_annotation.gff"
  vcf_dir_path: "data/11_vcf_output_for_R"
  sra_runtable_path: "data/00_sra_runtable/filtered_sra_table.txt"
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      error = FALSE)
```

# Background and Overview

This report examined how single nucleotide polymorphisms (SNPs) in different strains of SARS-CoV-2 affects their permeability and transmissibility. It has been established through the taken BioProject that the substitution S:655Y enhances replication of the disease, transmitting more efficiently than variants. The primary goal of this analysis were to identify the SNPs present in the variants of the disease and analyze how might the SNPs enhance the transmissibility of the variants. The approach was conducted through a series of bash scripts and Rstudio code. The bash scripts ultimately converted the metadata for the SRA run table to vcf files, which were made accessible to interpret. The analysis was possible through the data contained in the vcf file. 

(INSERT FINDINGS)
Mock Findings:
state what the ref nuc and alt nuc is. State what chrom pos and what the snp is. state the quality of score of each chrom pos.

maybe add as of march 22 gamma variant died (maybe got out competed
maybe add how gamma was prevalent in brazil and associated with increased death risk and severity in younger ages.
flow - snp lead to increased permeability to increased death

# Methods

See the set of tutorials on the [vcfR package website](https://knausb.github.io/vcfR_documentation/index.html).

You may also want to use any of a range of different COVID data packages and data sources:

* https://kjhealy.github.io/covdata/
* https://github.com/como-ph/oxcovid19
* https://ropensci.org/blog/2020/10/20/searching-medrxivr-and-biorxiv-preprint-data/
* https://covidtracking.com/data/api
  * `readr::read_csv("https://api.covidtracking.com/v1/states/daily.csv")`

## Subsections are ok too

# Results and Discussion

```{r load-packages-and-functions}
library("vcfR")
library("ggplot2")
library("dplyr")
library("ggthemes")
library("tidyr")

# this loads all function files in the code/functions directory
function_paths <- list.files("code/functions", full.names = TRUE)
purrr::walk(function_paths, source)
```

```{r load-vcf-data}
# load in, tidy, and stack all of the VCF files in the target directory
# these are the output of the bash pipeline
stacked_vcfs <- parse_tidy_and_stack_vcfs(
  vcf_dir_path = params$vcf_dir_path)

# load in the gff file with genome annotations for SARS-CoV-2
gff <- read_gff(gff_file_path = params$gff_file_path)

# pull out just the named genes from the annotation file
gene_table <- extract_genes_from_gff(annotation_object = gff)

# combine the stacked vcf table with the gene names and the metadata from
# the sra runtable file
vcf_with_metadata <- add_genes_metadata_to_vcfstack(
  sra_runtable_path = params$sra_runtable_path,
  stacked_vcf = stacked_vcfs,
  cleaned_genes_table = gene_table)
vcf_with_metadata$qual <- as.integer(vcf_with_metadata$qual)
colnames(vcf_with_metadata) <- gsub(" ", "_", colnames(vcf_with_metadata))
```

# Figures
```{r counts-plot}
# create a counts plot counting the quality scores of each sample
counts_plot <- ggplot(data = vcf_with_metadata,
                    aes(x = sample,
                        y = qual)) +
  geom_count(col = "tomato3", show.legend = F) +
  labs(x = "Sample", y = "Quality Score",
       title = "Quality score of each sample") +
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(axis.text.x = element_text(angle = 90, size = 5))

ggsave(plot = counts_plot,
       filename = paste0("output/counts_plot.png"))

counts_plot
```

**Figure 1** The different samples and the quality scores of their SNPs.

# Tables
```{r sample-bases-table}
# create a table showing the samples and how many bases they have and what the
# test sample is from

sample_bases_table <- vcf_with_metadata %>% select(sample, Bases, Isolation_source) %>%
  unique()

sample_bases_table
```

**Table 1** All of the samples, their base length, and isolation source

```{r SNP-table}
# create a table showing the samples and their SNP position along with the reference nucleotide

SNP_table <- stacked_vcfs %>%
  select(sample, ref, alt, pos) %>% reshape(stacked_vcfs, direction = "wide", idvar = "sample", timevar = "pos", varying = NULL, v.names = NULL, drop = NULL) %>%
  knitr::kable(col.names = c("Sample", "Ref. nucleotide 8782", "SNP 8782",
                             "Ref. nucleotide 18060", "SNP 18060", "Ref. nucleotide 23525", "SNP 23525", "Ref. nucleotide 28144", "SNP 28144", "Ref. nucleotide 29737", "SNP 29737", "Ref. nucleotide 18883", "SNP 18883"), align = "lcccccccccccc")

SNP_table
```

**Table 1**: Each sample, position, and the reference nucleotide and SNP

# Sources Cited
